pkg/cmd/agent/list.go

```diff
@@ -0,0 +1,77 @@
+package agent
+
+import (
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/buildkite/cli/v3/internal/io"
+	"github.com/buildkite/cli/v3/pkg/cmd/factory"
+	"github.com/charmbracelet/bubbles/table"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	"github.com/spf13/cobra"
+)
+
+func NewCmdAgentList(f *factory.Factory) *cobra.Command {
+	cmd := cobra.Command{
+		DisableFlagsInUseLine: true,
+		Use:                   "list",
+		Args:                  cobra.NoArgs,
+		Short:                 "List agents",
+		Long: heredoc.Doc(`
+			List all connected agents for the current organization.
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			l := io.NewPendingCommand(func() tea.Msg {
+				agents, _, err := f.RestAPIClient.Agents.List(f.Config.Organization, nil)
+				if err != nil {
+					return err
+				}
+
+				if len(agents) == 0 {
+					return io.PendingOutput("No connected agents")
+				}
+
+				var rows []table.Row
+				maxID, maxName, maxStatus, maxTags := 0, 0, 0, 0
+				for _, agent := range agents {
+					if max := len(*agent.ID); max > maxID {
+						maxID = max
+					}
+					if max := len(*agent.Name); max > maxName {
+						maxName = max
+					}
+					if max := len(*agent.ConnectedState); max > maxStatus {
+						maxStatus = max
+					}
+					tags := strings.Join(agent.Metadata, ", ")
+					if max := len(tags); max > maxTags {
+						maxTags = max
+					}
+					row := []string{
+						*agent.ID, *agent.Name, *agent.ConnectedState, tags,
+					}
+
+					rows = append(rows, row)
```

Comment:
An alternative solution to reduce the repetition might be something like:

```go
var rows []table.Row
maxLengths := map[string]int{"ID": 0, "Name": 0, "Status": 0, "Tags": 0}

for _, agent := range agents {
    agentData := map[string]string{
        "ID":     *agent.ID,
        "Name":   *agent.Name,
        "Status": *agent.ConnectedState,
        "Tags":   strings.Join(agent.Metadata, ", "),
    }

    // Update maximum lengths
    for key, value := range agentData {
        if len(value) > maxLengths[key] {
            maxLengths[key] = len(value)
        }
    }

    rows = append(rows, []string{agentData["ID"], agentData["Name"], agentData["Status"], agentData["Tags"]})
}
```

No real preference, I just initially didn't grasp what the `maxX` were until I saw the `{Title: "ID", Width: maxID}` lines below.

Reply:
Thats a bit nicer. I was just shoehorning this in for now since I'd like to reimplement the list output completely in some future tickets. This sort of thing should just be handled for us with that. Still looking into what that would look like though